external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -32,10 +33,31 @@ impl<'env, 'lexer, 'input> Context<'env, 'lexer, 'input> {
         tokens: &'lexer mut Lexer<'input>,
         package_name: Option<Symbol>,
     ) -> Self {
+        let mut stop_set = TokenSet::new();
+        stop_set.add(Tok::EOF);
         Self {
             package_name,
             env,
             tokens,
+            stop_set,
+        }
+    }
+
+    /// Checks if the current token is a member of the stop set.
+    fn at_stop_set(&self) -> bool {
+        self.tokens.at_set(&self.stop_set)
+    }
+
+    /// Advances tokens until reaching an element of the stop set, recording diagnostics along the
+    /// way (including the first optional one passed as an argument.
+    fn advance_until_at_stop_set(&mut self, diag_opt: Option<Diagnostic>) {
+        if let Some(diag) = diag_opt {
+            self.env.add_diag(diag);
+        }
```

Comment:
This feels a bit strange? Since it is always added? But maybe I will understand more as I read